src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -954,13 +915,6 @@ internal sealed class TupleExtraData
 
             private ImmutableArray<FieldSymbol> _lazyDefaultElementFields;
 
-            /// <summary>
-            /// For tuple fields that aren't TupleElementFieldSymbol or TupleErrorFieldSymbol, we cache their tuple element index.
-            /// This supports <see cref="FieldSymbol.TupleElementIndex"/>.
-            /// For those fields, we map from their definition to an index.
-            /// </summary>
-            private SmallDictionary<FieldSymbol, int>? _lazyFieldDefinitionsToIndexMap;
```

Comment:
>_lazyFieldDefinitionsToIndexMap [](http://example.com/codeflow?start=55&length=31)

Are we certain that nothing else in this class is prone to a similar race? Would it be simpler and more robust to ensure that members and tuple data are in sync by using an alternative approach? For example, by ensuring that they are stored together at once.

Reply:
I've reviewed the other parts of `TupleExtraData` and didn't spot a similar race condition. 
That said, there is a race condition left somewhere (issue https://github.com/dotnet/roslyn/issues/53943) which seems different to the one fixed here. I haven't identified the root cause yet.

For the suggestion to have `MakeSynthesizedTupleMembers` return the map and have the caller store it, we'd discussed this with Chuck among a few options before the holiday break (but we leaned towards removing the map). I'm open to the approach if Chuck is okay with it. 
If we do go that route, do you have an initialization pattern for the caller? Feels like we'd need to spin-wait the threads that are waiting for the winning thread to save the map.